### PR TITLE
Add delegation controller

### PR DESCRIPTION
**Pull request by danielailie**

### CODE REVIEW

File: src/delegation/delegationController.ts [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+        this.factory = new DelegationTransactionsFactory({
+            config: new TransactionsFactoryConfig({ chainID: options.chainID }),
+        });
+        this.parser = new DelegationTransactionsOutcomeParser();
+        this.txComputer = new TransactionComputer();
+    }
+
+    async createTransactionForNewDelegationContract(
+        sender: IAccount,
+        nonce: bigint,
```
Comment (andreibancioiu) [inline src/delegation/delegationController.ts:29]: :+1:

File: src/delegation/delegationController.ts [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+        options: inputs.NewDelegationContractInput,
```
Comment (andreibancioiu) [inline src/delegation/delegationController.ts:30]: Good naming convention.

File: src/delegation/delegationTransactionsFactory.ts [hunk 3]
```diff
@@ -20,7 +21,7 @@ interface IConfig {
     additionalGasLimitForDelegationOperations: bigint;
 }
 
-interface IValidatorPublicKey {
+export interface IValidatorPublicKey {
```
Comment (andreibancioiu) [inline src/delegation/delegationTransactionsFactory.ts:24]: Should not be exported, this is an internal interface. Can be handled as the other re-defined interfaces.

File: src/delegation/delegationTransactionsFactory.ts [hunk 4]
```diff
@@ -368,14 +330,36 @@ export class DelegationTransactionsFactory {
...
+        throw new Error("Method not implemented.");
+    }
+    createTransactionForRedelegatingRewards(
+        _sender: IAddress,
+        _options: inputs.ManageDelegationContractInput,
+    ): Transaction {
+        throw new Error("Method not implemented.");
+    }
+    createTransactionForClaimingRewards(
+        _sender: IAddress,
```
Comment (andreibancioiu) [inline src/delegation/delegationTransactionsFactory.ts:354]: Spacing issue / newlines.

File: src/delegation/delegationController.ts [hunk 5]
```diff
@@ -0,0 +1,288 @@
+import { IAccount } from "../accounts/interfaces";
+import { ITransactionOnNetwork } from "../interfaceOfNetwork";
+import { INetworkProvider } from "../networkProviders/interface";
+import { Transaction } from "../transaction";
+import { TransactionComputer } from "../transactionComputer";
+import { TransactionsFactoryConfig } from "../transactionsFactories";
+import { TransactionWatcher } from "../transactionWatcher";
+import { DelegationTransactionsFactory } from "./delegationTransactionsFactory";
+import { DelegationTransactionsOutcomeParser } from "./delegationTransactionsOutcomeParser";
+import * as inputs from "./resources";
```
Comment (andreibancioiu) [inline src/delegation/delegationController.ts:10]: `inputs vs. resources` - what if we also had outputs? Can also be simply, `resources`.

**Review by andreibancioiu — APPROVED** [review]

**Review by popenta — APPROVED** [review]

**Pull request merged**